lfs/client.go

```diff
@@ -595,3 +603,25 @@ func sendApiEvent(event int) {
 	default:
 	}
 }
+
+type notImplError struct {
+	error
+}
+
+func (e notImplError) NotImplemented() bool {
+	return true
+}
+
+func newNotImplError() error {
+	return notImplError{errors.New("Not Implemented")}
+}
+
+func isNotImplError(err *WrappedError) bool {
+	type notimplerror interface {
+		NotImplemented() bool
+	}
+	if e, ok := err.Err.(notimplerror); ok {
+		return e.NotImplemented()
+	}
```

Comment:
This is avoiding directly comparing error types, using this method http://dave.cheney.net/2014/12/24/inspecting-errors